### ISSUE TITLE
Lua API: coordinate with DBX

### ISSUE THREAD

**Issue by Gaerzi** [sirjuddington/SLADE#992]
Anotak's DBX also features a Lua API for scripting map editing. If at all possible, it would be a good idea to have the map editing Lua API in SLADE similar enough that scripts written for one could be used in the other.

For reference, here is DBX's documentation. [https://github.com/anotak/doombuilderx/wiki](https://github.com/anotak/doombuilderx/wiki)

**Comment by sirjuddington** [comment]
Yeah that's something I was going to try to do when I get around to doing more of the map editor scripting stuff. Not sure how much I'll be able to keep the same, it might be necessary to create alternate lua types for dbx compatibility.

**Comment by anotak** [comment]
i thought about this early on and i had kind of decided it was prohibitively difficult to implement on the editor side due to structural differences.

i thought the best approach might be to write wrappers in lua itself, that could be require/dofile'd from a script. but i didn't investigate it too thoroughly.

**Comment by zgramps** [comment]
The Pen and some of the UI stuff look very useful. Most of the the other stuff does look like it could be shimmed in Lua.

**Comment by sirjuddington** [comment]
I'll add the link to more places once the scripting system is closer to being complete
